app/javascript/layouts/Layout.tsx

```diff
@@ -31,12 +34,33 @@ const Layout = (props: LayoutProps) => {
 
   const currentPath = window.location.pathname
 
+  let notice = null
+  if (process.env.SHOW_RESEARCH_BANNER) {
+    notice = (
+      <Markdown>
+        {t("nav.researchFeedback", { researchUrl: process.env.RESEARCH_FORM_URL })}
+      </Markdown>
+    )
+  }
+  let topAlert = null
+  if (process.env.TOP_MESSAGE) {
+    topAlert = (
+      <AlertBox
+        type={(process.env.TOP_MESSAGE_TYPE as AlertTypes) || "alert"}
```

Comment:
I'm a little worried about using `as` here. I'm reading a book on typescript right now and they say to avoid it where possible. Like I'm not sure the implementation of AlertBox or what would happen if a non-AlertTypes string was passed in.

You could add a small top-level function to coerce it for you:

```
const asAlertType = (alertType: string): AlertTypes => {
  switch (alertType) {
    case "notice":
      return "notice"
    case "success":
      return "success"
    case "alert":
    default:
      return "alert"
  }
}
```

and then here you'd do:
```
type={asAlertType(process.env.TOP_MESSAGE_TYPE)}
```

Reply:
that works, sure, will amend